### PR TITLE
asg/mem-alloc: Use doubly linked-list and update checker

### PR DESCRIPTION
**Pull request by Alex-deVis**
Add a pointer to prev in `struct block_meta` and move unmodifiable files to `utils/`.
Add memory leak checks and style checks to the checker.
Add `make check-fast` option to skip checking for memory leaks.
Add instructions for debugging with the VSCode debugger.

### CODE REVIEW

File: content/assignments/memory-allocator/README.md [hunk 1]
```diff
@@ -234,58 +240,46 @@ test-realloc-coalesce            ........................ passed ...   3
 test-realloc-coalesce-big        ........................ passed ...   1
 test-all                         ........................ passed ...   5
 
-Grade                            .................................. 9.00
+Total:                                                           100/100
```
Comment (razvand) [inline content/assignments/memory-allocator/README.md:243]: The total has to be 90/100. Update the grading inside the tests.

File: content/assignments/memory-allocator/README.md [hunk 2]
```diff
@@ -234,58 +240,46 @@ test-realloc-coalesce            ........................ passed ...   3
...
 
-To run a single test use `checker.py <test>`, where `<test>` is the name of the test without path or extension.
-Append `-v` to see the diff between `out/<test>.out` and `ref/<test>.ref`.
+`run-tests.py` supports three modes:
+
+- verbose (`-v`), prints the output of the test
+- diff (`-d`), prints the diff between the output and the ref
+- memcheck (`-m`), prints the diff between the output and the ref and announces memory leaks
+
+If you want to run a single test, you give its name or its path as arguments to `run-tests.py`
```
Comment (razvand) [inline content/assignments/memory-allocator/README.md:263]:
```suggestion
If you want to run a single test, you give its name or its path as arguments to `run-tests.py`:
```

**Review by razvand — CHANGES_REQUESTED** [review]
Why is `checkpatch.pl` part of the repository. `checkpatch.pl` is to be used in conjunction with the Dockerfile as part of the [GitLab assignment repositor](). No need to add it here.

Ignore resulting executables in `snippets/`.

**Comment by Alex-deVis** [comment]
> Ignore resulting executables in snippets/.

#TIL that you can write a `.gitignore` that ignores itself.

**Comment by Alex-deVis** [comment]
@razvand bump on this

**Comment by teodutu** [comment]
@razvand @Alex-deVis, thanks to @gabrielmocanu's arduous work, we can now render PRs that make changes to the website by adding the `needs-rendering` label. Use it whenever you need it. The action posts a comment with a link to the updated website.

**Review by razvand — APPROVED** [review]
All good.

**Pull request merged**